libs/api/transaction/data-access/src/lib/api-transaction-data-access.service.ts

```diff
@@ -111,4 +111,18 @@ export class ApiTransactionDataAccessService {
   sendVerifyWebhook(app: App, payload: Prisma.AppTransactionUpdateInput) {
     return this.appWebhook.sendWebhook(app, { type: AppWebhookType.Verify, payload })
   }
+
+  parseError(error) {
+    const description = error.toString()
+    if (description.includes('invalid account data for instruction')) {
+      const instructionIndex = description.split(' ')[11]
```

Comment:
While this works, I am not a huge fan of doing validations on strings. However that's what we have so far coming from Solana. I will check if I can do a better job when working on #200

Reply:
It seems that what's [used under the hoods](https://github.com/solana-labs/solana-web3.js/blob/4c75263c3180866168a4f8a9826a96eb70a6dca1/src/connection.ts#L4218) returns a proper object (`SendTransactionError`) which might be a better way to handle it instead of string parsing.

Reply:
It won't work for what is intended here because the `SendTransactionError` it is thrown for any kind of error when sending a transaction; if for example it's an invalid accountId, it throws this, but if it is Insufficient funds it also throws this one.